skin/src/main/java/org/researchstack/skin/task/SignUpTask.java

```diff
@@ -57,60 +57,60 @@ private void initSteps(Context context) {
                 .create(context);
 
         for(InclusionCriteriaModel.Step s: model.steps) {
-            switch(s.type)
-            {
-                case INSTRUCTION:
-                    Step instruction = null;
-                    switch(s.identifier)
-                    {
-                        case InclusionCriteriaModel.INELIGIBLE_INSTRUCTION_IDENTIFIER:
-                            instruction = new Step(SignUpIneligibleStepIdentifier, s.text);
-                            instruction.setText(s.detailText);
-                            instruction.setStepTitle(R.string.rss_eligibility);
-                            instruction.setStepLayoutClass(SignUpIneligibleStepLayout.class);
-                            break;
-                        case InclusionCriteriaModel.ELIGIBLE_INSTRUCTION_IDENTIFIER:
-                            instruction = new Step(SignUpEligibleStepIdentifier, s.text);
-                            instruction.setText(s.detailText);
-                            instruction.setStepTitle(R.string.rss_eligibility);
-                            instruction.setStepLayoutClass(SignUpEligibleStepLayout.class);
-                            break;
-                        default:
-                            instruction.setStepTitle(R.string.rss_eligibility);
-                            instruction = new Step(s.identifier, s.text);
-                            instruction.setText(s.detailText);
-                    }
-
-                    stepMap.put(instruction.getIdentifier(), instruction);
-                    break;
-                // TODO: not sure what the differences are between compound/toggle or is compound obsolete?
-                case COMPOUND:
-                case TOGGLE:
-                    FormStep form = new FormStep(SignUpInclusionCriteriaStepIdentifier, s.text, s.detailText);
-                    List<QuestionStep> questions = new ArrayList<>();
-
-                    if(s.items != null)
-                    {
-                        // TODO: extend the json to include (yes/no)?
-                        BooleanAnswerFormat booleanAnswerFormat =
-                                new BooleanAnswerFormat(context.getString(R.string.rsb_yes), context.getString(R.string.rsb_no));
-                        for (InclusionCriteriaModel.Item item : s.items) {
-                            QuestionStep question = new QuestionStep(item.identifier, item.text, booleanAnswerFormat);
-                            answerMap.put(item.identifier, item.expectedAnswer);
-                            questions.add(question);
+            // step can be null with addition of new OnboardingManager steps, so just ignore them
+            if (s != null && s.type != null) {
```

Comment:
stupid git diff... all I did was add the conditional line, the rest is indentation.